### ISSUE TITLE
Pretraining using AutoEncoder in keras

### ISSUE THREAD

**Issue by sumo8291** [fchollet/keras#257]
i am trying to use pretraining process in my cnn
the code that i wrote is given below

model = Sequential()
    X_train=X_train.reshape(28709,1,48,48);
    model.add(AutoEncoder(encoder=Convolution2D(24, 1,7,7, border_mode='valid'),
                                     decoder=Convolution2D(1, 24,7,7, border_mode='full'),
                                     output_reconstruction=True, tie_weights=True))
    opt=SGD(lr=0.01, momentum=0., decay=0., nesterov=False)
    ### Compile the loss
    model.compile(loss='mean_squared_error', optimizer=opt)
    # Training Neural Network
    model.fit(X_train, X_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=2)

I m getting the following error

![screen shot 2015-06-21 at 4 27 54 pm](https://cloud.githubusercontent.com/assets/8149573/8271373/7c14c9ca-1831-11e5-8aa1-e0c75b10b2d8.png)

**Comment by fchollet** [comment]
It's not clear what the error means, but you targets appear to be a int8 array. Try using a float64 array instead (note that it might not be related to the error).
